src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -117,18 +118,29 @@ private T runWithRetries(byte[] key, int redirections, boolean tryRandomNode, bo
       }
 
       return execute(connection);
-    } catch (JedisConnectionException jce) {
-      if (tryRandomNode) {
-        // maybe all connection is down
-        throw jce;
-      }
 
+    } catch (JedisNoReachableClusterNodeException jnrcne) {
+      //Not sure if JedisNoReachableClusterNodeException exception should extends JedisConnectionException, it's not connection exception
+      //Remove this if hierarchy reworked and no need to make exception from JedisConnectionException clause
+      throw jnrcne;
+    } catch (JedisConnectionException jce) {
       // release current connection before recursion
       releaseConnection(connection);
       connection = null;
 
-      // retry with random connection
-      return runWithRetries(key, redirections - 1, true, asking);
+      if (redirections <= 1) {
```

Comment:
"Because as JedisCluster is today, if you sex maxRedirections to 1, you can get redirected only once." You will be 'directed' only once, but 'redirected' zero times. So, you make one 'attempt'. When I set maxRedirections to 1 I could assume that cluster client will make one request and follow one MOVED if required, when I set it to 0 I could assume that cluster client will make only one main attempt, but it doesn't work this way. (JedisClusterCommand#runWithRetries check ```redirections<=0``` to throw exception) In addition, maxRredirections doesn't include information that this count would be used as a number of attempts in case of any fail. A user could assume that it has only "MOVED" related semantic. There are my thoughts about this name.